cpp/src/gandiva/precompiled/arithmetic_ops.cc

```diff
@@ -220,10 +220,10 @@ CAST_UNARY_UTF8(castVARCHAR, int8, utf8, "%d", NOFMT, nothing)
 CAST_UNARY_UTF8(castVARCHAR, int16, utf8, "%d", NOFMT, nothing)
 CAST_UNARY_UTF8(castVARCHAR, int32, utf8, "%d", NOFMT, nothing)
 CAST_UNARY_UTF8(castVARCHAR, int64, utf8, "%ld", NOFMT, nothing)
-// CAST_UNARY_UTF8(castVARCHAR, float32, utf8, "%.*f", FMT, FLT_DIG)
-// CAST_UNARY_UTF8(castVARCHAR, float64, utf8, "%.*f", FMT, DBL_DIG)
-CAST_UNARY_UTF8(castVARCHAR, float32, utf8, "%g", NOFMT, nothing)
-CAST_UNARY_UTF8(castVARCHAR, float64, utf8, "%g", NOFMT, nothing)
+CAST_UNARY_UTF8(castVARCHAR, float32, utf8, "%.*f", FMT, FLT_DIG)
+CAST_UNARY_UTF8(castVARCHAR, float64, utf8, "%.*f", FMT, DBL_DIG)
```

Comment:
@zhouyuan Please take a look at this, thanks!

Reply:
can you please make a note here? this is required as Spark has some different behavior as gandiva

Reply:
note was added